Added JavaScript validation to contact form

anurag91920 opened the pull request:
## 🔖 PR Title:
<!-- A clear and concise title e.g. "Fixed leaderboard score update issue in PHP backend" -->

---

## Change ---

## ✨ Changes Introduced
- Added **dark mode toggle** (🌙 / ☀️).
- Improved **form validation**:
  - Inline error messages (no more alerts).
  - Debounced live validation for smoother UX.
  - Auto-trimming of email & phone inputs.
- Enhanced **UI/UX**:
  - Modern input styles with hover/focus effects.
  - Success message with auto-hide feature.
  - Better color contrast for accessibility.
- Added **dynamic subject field** (appears when "Other" is selected).

## ✅ Why This Change?
Earlier most contributions/issues were only about validation.  
This PR adds **new features + enhancements** to improve user experience and accessibility.

## 🔍 Testing Done
- Manually tested in both **light** and **dark** modes.
- Verified validations for name, email, phone, subject, and message.
- Confirmed success message works and form resets correctly.

## 📌 Notes
- Branch: `add-js-validation`
- Ready to be merged into `main`.


## Description
This pull request adds client-side validation to the contact form to ensure that users provide valid inputs before submission. The following validations have been implemented:

✅ Full Name is required

✅ Email format is validated to check if it contains @ and a domain

✅ Phone number is checked to ensure it is a 10-digit number

✅ Subject is required

✅ Message is required

✅ The form will not submit unless all fields are valid

✅ Error messages are shown near each invalid field

- [x] Bug Fix ✅
- [ ] New Feature ✨
- [x] Enhancement 🔧
- [ ] Documentation 📝
- [ ] UI/UX Update 🎨

---


## 📚 Documentation Updated?
- [x] Yes, I have updated the relevant section in `README.md` or related files.

---

## 🖼️ Screenshot or Screen Recording (max 15 seconds)
> (
<img width="1920" height="1080" alt="2025-09-08 (8)" src="https://github.com/user-attachments/assets/b1591696-3d86-40d8-88b3-3be8d5786eb1" />
).

## Vidio
[Vidio Link](https://drive.google.com/file/d/1GA2ncVVBPtnAp7zyRAVTwVX-75vmVy6j/view?usp=sharing)
---

## 💡 Tech Stack Used:
<!-- Mention technologies used -->
- HTML / CSS / JavaScript
- PHP / MySQL
- Tailwind / Bootstrap 
- Other:

---

## List of File Changed :

# no. of file changed :
<!-- write the names-->


## 🔗 Issue Reference:
<!-- Mention the issue number this PR fixes -->
Closes #IssueNumber [#397]

## Pr Number
#423 

---

## 🧩 Contribution Type:
- [ ] Frontend 🎯
- [ ] Backend 🔙
- [ ] UI/UX 🎨
- [ ] Documentation 📘
- [ ] Bug Fix 🐛
- [ ] Feature Implementation 🚀
- [ ] Optimization ♻️
- [ ] AI/ML
- [ ] Other:


---

## 🙏 Additional Info (if any):
Name: Anurag Chaurasiya
GitHub ID: [anurag91920](https://github.com/anurag91920)
LinkedIn Username (for appreciation): Anurag
LinkedIn Profile URL (for verification): [Click Here](https://www.linkedin.com/in/anurag-chaurasiya-32a2442a5)
Email ID (for communication): [anurag9120959628@gmail.com](mailto:anurag9120959628@gmail.com)
Participation Program: Open Source Connect

anurag91920 commented:
Please Merje the pr

VAIBHAVBABELE commented:
Not required.

VAIBHAVBABELE commented:
Sab log sirf validate hi karte h is ko lekar 7 + issue create ho chuke h. Koi naya feature add karo ya fr enhancement karo

anurag91920 commented:
## ✨ Changes Introduced
- Added **dark mode toggle** (🌙 / ☀️).
- Improved **form validation**:
  - Inline error messages (no more alerts).
  - Debounced live validation for smoother UX.
  - Auto-trimming of email & phone inputs.
- Enhanced **UI/UX**:
  - Modern input styles with hover/focus effects.
  - Success message with auto-hide feature.
  - Better color contrast for accessibility.
- Added **dynamic subject field** (appears when "Other" is selected).

## ✅ Why This Change?
Earlier most contributions/issues were only about validation.  
This PR adds **new features + enhancements** to improve user experience and accessibility.

## 🔍 Testing Done
- Manually tested in both **light** and **dark** modes.
- Verified validations for name, email, phone, subject, and message.
- Confirmed success message works and form resets correctly.

## 📌 Notes
- Branch: `add-js-validation`
- Ready to be merged into `main`.

anurag91920 commented:
please merge the pr

VAIBHAVBABELE commented:
https://discord.gg/AgEfwAkn join

anurag91920 commented:
please merge the pr
